### PR TITLE
REM-848: unit tests for pubic key handler functionality

### PR DESCRIPTION
**Pull request by dmytrostriletskyi**

### CODE REVIEW

File: remme/settings/__init__.py [hunk 1]
```diff
@@ -7,7 +7,9 @@
 SETTINGS_PUB_KEY_ENCRYPTION = 'remme.settings.pub_key_encryption'
 SETTINGS_KEY_GENESIS_OWNERS = 'remme.settings.genesis_owners'
 SETTINGS_SWAP_COMMISSION = 'remme.settings.swap_comission'
-SETTINGS_STORAGE_PUB_KEY = 'remme.settings.storage_pub_key'
+
+# Backward compatibility to use more obvious variable name
+SETTINGS_KEY_TO_GET_STORAGE_PUBLIC_KEY_BY = SETTINGS_STORAGE_PUB_KEY = 'remme.settings.storage_pub_key'
```
Comment (eugene-babichenko) [inline remme/settings/__init__.py]: It became even worse :) What's so wrong with the old one?

File: remme/tp/pub_key.py [hunk 2]
```diff
@@ -41,8 +46,12 @@
 FAMILY_VERSIONS = ['0.1']
 
 PUB_KEY_ORGANIZATION = 'REMME'
-PUB_KEY_MAX_VALIDITY = timedelta(365)
-PUB_KEY_STORE_PRICE = 10
+
+# Backward compatibility to use more obvious variables name
```
Comment (eugene-babichenko) [inline remme/tp/pub_key.py]: No. We don't operate on certificates here, blockchain knows nothing about them.

File: remme/tp/pub_key.py [hunk 3]
```diff
@@ -61,95 +70,167 @@ def get_state_processor(self):
...
                 sigerr += 1
 
         if sigerr == 2:
-            raise InvalidTransaction('Invalid signature')
+            return False
+
+        return True
+
+    @staticmethod
+    def _is_certificate_validity_exceeded(valid_from, valid_to):
```
Comment (eugene-babichenko) [inline remme/tp/pub_key.py]: Again, public keys.

File: remme/tp/pub_key.py [hunk 4]
```diff
@@ -61,95 +70,167 @@ def get_state_processor(self):
...
+
+        transfer_state = AccountHandler()._transfer_from_address(
+            context=context, address_from=address_from, transfer_payload=transfer_payload,
+        )
+
+        return transfer_state
+
+    def _store_pub_key(self, context, signer_public_key, new_public_key_payload):
+        """
+        Store certificate public key to the blockchain.
```
Comment (eugene-babichenko) [inline remme/tp/pub_key.py]: And again :)
This can be not only a certificate, but also any other container.

File: remme/tp/pub_key.py [hunk 5]
```diff
@@ -61,95 +70,167 @@ def get_state_processor(self):
...
-                    .create_transfer(context, account_address, storage_address,
-                                     PUB_KEY_STORE_PRICE)
+        state = {
+            sender_account_address: account,
+            public_key_to_store_address: public_key_to_store,
+        }
+
+        is_economy_enabled = _get_setting_value(context, 'remme.economy_enabled', 'true').lower()
+
+        if is_economy_enabled == ECONOMY_IS_ENABLED_VALUE:
```
Comment (eugene-babichenko) [inline remme/tp/pub_key.py]: A little note about code style: we don't use an empty newline after `if` statement.

File: remme/settings/__init__.py (hunk 1 above)
Comment (dmytrostriletskyi) [reply remme/settings/__init__.py]: @eugene-babichenko, `SETTINGS_STORAGE_PUB_KEY` says it is real public key, not key to get public key by.

File: remme/tp/pub_key.py (hunk 2 above)
Comment (dmytrostriletskyi) [reply remme/tp/pub_key.py]: @eugene-babichenko, so `CERTIFICATE_PUBLIC_KEY_MAXIMUM_VALIDITY ` to `PUBLIC_KEY_MAXIMUM_VALIDITY`?

File: remme/tp/pub_key.py (hunk 3 above)
Comment (dmytrostriletskyi) [reply remme/tp/pub_key.py]: @eugene-babichenko, got you.

File: remme/tp/pub_key.py (hunk 4 above)
Comment (dmytrostriletskyi) [reply remme/tp/pub_key.py]: @eugene-babichenko, got you.

File: remme/tp/pub_key.py (hunk 5 above)
Comment (dmytrostriletskyi) [reply remme/tp/pub_key.py]: @eugene-babichenko, got you.

**Pull request merged**